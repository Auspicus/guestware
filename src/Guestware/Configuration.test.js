jest.mock('node-fetch')

import { Configuration } from '../../lib/index.js'

describe('Configuration', () => {

  const secrets = {
    wsdl: 'wsdl',
    username: 'username',
    password: 'password',
    appName: 'appName',
    version: 'version'
  }
  const configuration = new Configuration(secrets)
  let xmlRequest = `
    <soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:web="http://webservices.guestware.com/">
      <soapenv:Header>
        <web:GWCNOBJ>
          <web:UserName>{{UserName}}</web:UserName>
          <web:PassWord>{{PassWord}}</web:PassWord>
          <web:ApplicationName>{{AppName}}</web:ApplicationName>
          <web:VersionNumber>{{Version}}</web:VersionNumber>
        </web:GWCNOBJ>
      </soapenv:Header>
      <soapenv:Body>
        <web:ReadGuestLogin>
            <web:parintGuestID>1234567890</web:parintGuestID>
        </web:ReadGuestLogin>
      </soapenv:Body>
    </soapenv:Envelope>`

  test('.apply => returns XML request with configuration applied', () => {
    xmlRequest = configuration.apply(xmlRequest)
    expect(xmlRequest).toMatchSnapshot()
  })

  test('.unapply => returns XML request with configuration removed', () => {
    xmlRequest = configuration.unapply(xmlRequest)
    expect(xmlRequest).toMatchSnapshot()
  })

})