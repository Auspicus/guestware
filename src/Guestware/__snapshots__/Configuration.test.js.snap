// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configuration .apply => returns XML request with configuration applied 1`] = `
"
    <soapenv:Envelope xmlns:soapenv=\\"http://schemas.xmlsoap.org/soap/envelope/\\" xmlns:web=\\"http://webservices.guestware.com/\\">
      <soapenv:Header>
        <web:GWCNOBJ>
          <web:UserName>username</web:UserName>
          <web:PassWord>password</web:PassWord>
          <web:ApplicationName>appName</web:ApplicationName>
          <web:VersionNumber>version</web:VersionNumber>
        </web:GWCNOBJ>
      </soapenv:Header>
      <soapenv:Body>
        <web:ReadGuestLogin>
            <web:parintGuestID>1234567890</web:parintGuestID>
        </web:ReadGuestLogin>
      </soapenv:Body>
    </soapenv:Envelope>"
`;

exports[`Configuration .unapply => returns XML request with configuration removed 1`] = `
"
    <soapenv:Envelope xmlns:soapenv=\\"http://schemas.xmlsoap.org/soap/envelope/\\" xmlns:web=\\"http://webservices.guestware.com/\\">
      <soapenv:Header>
        <web:GWCNOBJ>
          <web:UserName>{{UserName}}</web:UserName>
          <web:PassWord>{{PassWord}}</web:PassWord>
          <web:ApplicationName>{{AppName}}</web:ApplicationName>
          <web:VersionNumber>{{Version}}</web:VersionNumber>
        </web:GWCNOBJ>
      </soapenv:Header>
      <soapenv:Body>
        <web:ReadGuestLogin>
            <web:parintGuestID>1234567890</web:parintGuestID>
        </web:ReadGuestLogin>
      </soapenv:Body>
    </soapenv:Envelope>"
`;
